### PR TITLE
Okta new rules

### PR DESCRIPTION
**Pull request by arielkr256**
### Background

Okta is a centralized identity platform that serves as a modern alternative to traditional on-premises Domain Controllers for organizations. It provides a secure and unified way to manage user authentication, access control, and identity management across various applications and services. While the increased centralization of access controls can protect end users, it also presents a high-value target for attackers.

In a keynote address at the mWISE Conference 2023, Kevin Mandia stated that defenders must assume phishing attacks will be successful on their employees and that valid accounts will be used for lateral movement by attackers. Recent attacks, such as the scattered spider attacks on MGM and Caesar's, have confirmed these concerns. These attacks involved social engineering and phishing techniques to gain access to legitimate Okta admin accounts.

The Okta security team has emphasized that there were no vulnerabilities in the Okta platform that were exploited in these attacks. Instead, the attackers took advantage of misconfigurations and a lack of security controls to leverage compromised accounts into additional access, privileges, and persistence. It is crucial for organizations to closely monitor and detect new Tactics, Techniques, and Procedures (TTPs) in the Okta audit logs to identify any suspicious activities and potential security breaches.

### Changes

* 7 new detections based on recent high profile attacks by threat actors including LAPSU$, Scatter Swine, and Scattered Spider

### Testing

* sample logs are included for unit tests

### CODE REVIEW

File: rules/okta_rules/okta_anonymizing_vpn_login.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
+AnalysisType: rule
+Filename: okta_anonymizing_vpn_login.py
+RuleID: "Okta.Anonymizing.VPN.Login"
+DisplayName: "Okta Sign-In from VPN Anonymizer"
+Enabled: true
+LogTypes:
+    - Okta.SystemLog
+# Tags:
```
Comment (darwayne) [inline rules/okta_rules/okta_anonymizing_vpn_login.yml:8]: `[nit]` we should remove this

File: rules/okta_rules/okta_anonymizing_vpn_login.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+Severity: High
+Description: >
+    A user is attempting to sign-in to Okta from a known VPN anonymizer.  The threat actor would access the compromised account using anonymizing proxy services. 
+Runbook: >
+    Restrict this access to trusted Network Zones and deny access from anonymizing proxies in policy using a Dynamic Network Zone.
+Reference: >
+    https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection
+DedupPeriodMinutes: 30
+Threshold: 1
+# SummaryAttributes:
```
Comment (darwayne) [inline rules/okta_rules/okta_anonymizing_vpn_login.yml:21]: `[nit]` we should remove this

File: rules/okta_rules/okta_idp_create_modify.yaml [hunk 3]
```diff
@@ -0,0 +1,167 @@
+AnalysisType: rule
+Filename: okta_idp_create_modify.py
+RuleID: "Okta.Identity.Provider.Created.Modified"
+DisplayName: "Okta Identity Provider Created or Modified"
+Enabled: true
+LogTypes:
+    - Okta.SystemLog
+# Tags:
```
Comment (darwayne) [inline rules/okta_rules/okta_idp_create_modify.yaml:8]: `[nit]` we should remove this

File: rules/okta_rules/okta_idp_create_modify.yaml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    (sometimes called “Org2Org”) with the target.
+Runbook: >
+    Delegate access to this feature to a Custom Admin Role with the minimum required permissions.
+    Constrain these roles to groups that exclude highly privileged administrators.
+Reference: >
+    https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection
+    https://developer.okta.com/docs/concepts/identity-providers/
+DedupPeriodMinutes: 30
+Threshold: 1
+# SummaryAttributes:
```
Comment (darwayne) [inline rules/okta_rules/okta_idp_create_modify.yaml:29]: `[nit]` we should remove this

File: rules/okta_rules/okta_idp_signin.yaml [hunk 5]
```diff
@@ -0,0 +1,169 @@
+AnalysisType: rule
+Filename: okta_idp_signin.py
+RuleID: "Okta.Identity.Provider.SignIn"
+DisplayName: "Okta Identity Provider Sign-in"
+Enabled: false
+LogTypes:
+    - Okta.SystemLog
+# Tags:
```
Comment (darwayne) [inline rules/okta_rules/okta_idp_signin.yaml:8]: `[nit]` we should remove this

File: rules/okta_rules/okta_idp_signin.yaml [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    A user has signed in using a 3rd party Identity Provider.  
+    Attackers have been observed configuring a second Identity Provider to act as an "impersonation app" 
+    to access applications within the compromised Org on behalf of other users. This second Identity Provider, 
+    also controlled by the attacker, would act as a “source” IdP in an inbound federation relationship 
+    (sometimes called “Org2Org”) with the target.
+    From this “source” IdP, the threat actor manipulated the username parameter for targeted users in the second 
+    “source” Identity Provider to match a real user in the compromised “target” Identity Provider. 
+    This provided the ability to Single sign-on (SSO) into applications in the target IdP as the targeted user.
+    Do not use this rule if your organization uses legitimate 3rd-party Identity Providers.
+# Runbook: >
```
Comment (darwayne) [inline rules/okta_rules/okta_idp_signin.yaml:24]: `[nit]` we should remove this

File: rules/okta_rules/okta_idp_signin.yaml [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+Reference: >
+    https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection
+    https://developer.okta.com/docs/concepts/identity-providers/
+DedupPeriodMinutes: 30
+Threshold: 1
+# SummaryAttributes:
```
Comment (darwayne) [inline rules/okta_rules/okta_idp_signin.yaml:30]: `[nit]` we should remove this

File: rules/okta_rules/okta_new_behavior_accessing_admin_console.yml [hunk 8]
```diff
@@ -0,0 +1,254 @@
+AnalysisType: rule
+Filename: okta_new_behavior_accessing_admin_console.py
+RuleID: "Okta.New.Behavior.Accessing.Admin.Console"
+DisplayName: "Okta New Behaviors Acessing Admin Console"
+Enabled: true
+LogTypes:
+    - Okta.SystemLog
+# Tags:
```
Comment (darwayne) [inline rules/okta_rules/okta_new_behavior_accessing_admin_console.yml:8]: `[nit]` we should remove this

File: rules/okta_rules/okta_new_behavior_accessing_admin_console.yml [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+    A user attempted to access the Okta Admin Console from a new device with a new IP.
+Runbook: >
+    Configure Authentication Policies (Application Sign-on Policies) for access to privileged applications, including the Admin Console, to require re-authentication “at every sign-in”.
+    Turn on and test New Device and Suspicious Activity end-user notifications.
+Reference: >
+    https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection
+    https://help.okta.com/en-us/Content/Topics/Security/Security_General.htm
+DedupPeriodMinutes: 60
+Threshold: 1
+# SummaryAttributes:
```
Comment (darwayne) [inline rules/okta_rules/okta_new_behavior_accessing_admin_console.yml:24]: `[nit]` we should remove this

File: rules/okta_rules/okta_org2org_creation_modification.yml [hunk 10]
```diff
@@ -0,0 +1,255 @@
+AnalysisType: rule
+Filename: okta_org2org_creation_modification.py
+RuleID: "Okta.Org2org.Creation.Modification"
+DisplayName: "Okta Org2Org application created of modified"
+Enabled: true
+LogTypes:
+    - Okta.SystemLog
+# Tags:
```
Comment (darwayne) [inline rules/okta_rules/okta_org2org_creation_modification.yml:8]: `[nit]` we should remove this

File: rules/okta_rules/okta_org2org_creation_modification.yml [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+    MITRE ATT&CK:
+        - TA0006:T1556 # Modify Authentication Process
+        - TA0004:T1078.004 # Valid Accounts: Cloud Accounts
+Severity: High
+Description: >
+    An Okta Org2Org application has been created or modified.
+    Okta's Org2Org applications instances are used to push and match users from one Okta organization to another.
+    A malicious actor can add an Org2Org application instance and create a user in the source organization (controlled by the attacker) 
+    with the same identifier as a Super Administrator in the target organization.
+# Runbook: >
```
Comment (darwayne) [inline rules/okta_rules/okta_org2org_creation_modification.yml:19]: `[nit]` we should remove this

File: rules/okta_rules/okta_org2org_creation_modification.yml [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+Reference: >
+    https://www.authomize.com/blog/authomize-discovers-password-stealing-and-impersonation-risks-to-in-okta/
+    https://cardinalops.com/blog/splunk-and-other-siem-detections-for-okta-passbleed/
+DedupPeriodMinutes: 60
+Threshold: 1
+# SummaryAttributes:
```
Comment (darwayne) [inline rules/okta_rules/okta_org2org_creation_modification.yml:25]: `[nit]` we should remove this

File: rules/okta_rules/okta_password_extraction_via_scim.yaml [hunk 13]
```diff
@@ -0,0 +1,161 @@
+AnalysisType: rule
+Filename: okta_password_extraction_via_scim.py
+RuleID: "Okta.Password.Extraction.via.SCIM"
+DisplayName: "Okta Cleartext Passwords Extracted via SCIM Application"
+Enabled: true
+LogTypes:
+    - Okta.SystemLog
+# Tags:
```
Comment (darwayne) [inline rules/okta_rules/okta_password_extraction_via_scim.yaml:8]: `[nit]` we should remove this

File: rules/okta_rules/okta_password_extraction_via_scim.yaml [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+Reports:
+    MITRE ATT&CK:
+        - TA0006:T1556 # Modify Authentication Process
+Severity: High
+Description: >
+    An application admin has extracted cleartext user passwords via SCIM app.
+    Malcious actors can extract plaintext passwords by creating a SCIM application under their control and configuring it to sync passwords from Okta. 
+# Runbook: >
```
Comment (darwayne) [inline rules/okta_rules/okta_password_extraction_via_scim.yaml:16]: `[nit]` we should remove this

File: rules/okta_rules/okta_password_extraction_via_scim.yaml [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+Reference: >
+    https://www.authomize.com/blog/authomize-discovers-password-stealing-and-impersonation-risks-to-in-okta/
+    https://cardinalops.com/blog/splunk-and-other-siem-detections-for-okta-passbleed/
+DedupPeriodMinutes: 30
+Threshold: 1
+# SummaryAttributes:
```
Comment (darwayne) [inline rules/okta_rules/okta_password_extraction_via_scim.yaml:22]: `[nit]` we should remove this

File: rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.yml [hunk 16]
```diff
@@ -0,0 +1,166 @@
+AnalysisType: rule
+Filename: okta_phishing_attempt_blocked_by_fastpass.py
+RuleID: "Okta.Phishing.Attempt.Blocked.FastPass"
+DisplayName: "Okta AiTM Phishing Attempt Blocked by FastPass"
+Enabled: true
+LogTypes:
+    - Okta.SystemLog
+# Tags:
```
Comment (darwayne) [inline rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.yml:8]: `[nit]` we should remove this

File: rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.yml [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+Runbook: >
+    Protect sign-in flows by enforcing phishing-resistant authentication with Okta FastPass and FIDO2 WebAuthn.
+Reference: >
+    https://sec.okta.com/fastpassphishingdetection
+    https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection
+    https://sec.okta.com/sessioncookietheft
+    https://sec.okta.com/phishingasaservice
+DedupPeriodMinutes: 30
+Threshold: 1
+# SummaryAttributes:
```
Comment (darwayne) [inline rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.yml:26]: `[nit]` we should remove this

**Review by darwayne — APPROVED** [review]
LGTM, had some minor nit picks

**Comment by arielkr256** [comment]
cleaned up some unused yaml fields

File: rules/okta_rules/okta_idp_signin.py [hunk 18]
```diff
@@ -0,0 +1,17 @@
+from panther_base_helpers import okta_alert_context
+
+
+def rule(event):
+    return event.get("eventType") == "user.authentication.auth_via_IDP"
```
Comment (egibs) [inline rules/okta_rules/okta_idp_signin.py:5]: Same comment re: default.

File: rules/okta_rules/okta_new_behavior_accessing_admin_console.py [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    )
+
+    if "Okta Admin Console" not in target_app_names:
+        return False
+
+    behaviors = deep_get(event, "debugContext", "debugData", "behaviors")
+    if behaviors:
+        return "New Device=POSITIVE" in behaviors and "New IP=POSITIVE" in behaviors
+
+    return (
```
Comment (egibs) [inline rules/okta_rules/okta_new_behavior_accessing_admin_console.py:19]: Same comment re: default for both `deep_get` calls in the `return`.

File: rules/okta_rules/okta_new_behavior_accessing_admin_console.py [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+        == "POSITIVE"
+        and deep_get(
+            event, "debugContext", "debugData", "logOnlySecurityData", "behaviors", "New IP"
+        )
+        == "POSITIVE"
+    )
+
+
+def title(event):
+    return (
```
Comment (egibs) [inline rules/okta_rules/okta_new_behavior_accessing_admin_console.py:32]: Same comment re: default.

File: rules/okta_rules/okta_org2org_creation_modification.py [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+
+APP_LIFECYCLE_EVENTS = (
+    "application.lifecycle.update",
+    "application.lifecycle.create",
+    "application.lifecycle.activate",
+)
+
+
+def rule(event):
+    if event.get("eventType") not in APP_LIFECYCLE_EVENTS:
```
Comment (egibs) [inline rules/okta_rules/okta_org2org_creation_modification.py:11]: Same comment re: default.

File: rules/okta_rules/okta_org2org_creation_modification.py [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+        return False
+
+    target_app_names = get_val_from_list(
+        event.get("target", [{}]), "displayName", "type", "AppInstance"
```
Comment (egibs) [inline rules/okta_rules/okta_org2org_creation_modification.py:15]: Same comment re: `deep_walk`.

File: rules/okta_rules/okta_org2org_creation_modification.py [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+def title(event):
+    action = event.get("eventType").split(".")[2]
+    return (
+        f"Okta: [{event.get('actor',{}).get('alternateId','<id-not-found>')}] "
+        f"{action} Org2Org application"
+    )
+
+
+def severity(event):
+    if "create" in event.get("eventType"):
```
Comment (egibs) [inline rules/okta_rules/okta_org2org_creation_modification.py:33]: Same comment re: default.

File: rules/okta_rules/okta_password_extraction_via_scim.py [hunk 24]
```diff
@@ -0,0 +1,20 @@
+from panther_base_helpers import deep_get, okta_alert_context
+
+
+def rule(event):
+    return event.get(
+        "eventType"
+    ) == "application.lifecycle.update" and "Pushing user passwords" in deep_get(
+        event, "outcome", "reason"
```
Comment (egibs) [inline rules/okta_rules/okta_password_extraction_via_scim.py:8]: Same comment re: default.

File: rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.py [hunk 25]
```diff
@@ -0,0 +1,20 @@
+from panther_base_helpers import deep_get, okta_alert_context
+
+
+def rule(event):
+    return (
```
Comment (egibs) [inline rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.py:5]: Same comment re: default for `get` and both `deep_get` calls.

File: rules/okta_rules/okta_idp_create_modify.py [hunk 26]
```diff
@@ -0,0 +1,23 @@
+from panther_base_helpers import okta_alert_context
+
+
+def rule(event):
+    return "system.idp.lifecycle" in event.get("eventType")
```
[the remark this answers is not in the document]
Comment (arielkr256) [reply rules/okta_rules/okta_idp_create_modify.py:5]: If eventType is empty rule() returns False which is expected behavior.  Also for Okta logs eventType should never be empty.

File: rules/okta_rules/okta_idp_signin.py (hunk 18 above)
Comment (arielkr256) [reply rules/okta_rules/okta_idp_signin.py:5]: I agree with default strings for human readable things like title, but for boolean logic if 'eventType' doesn't exist, then default value is None and the rule returns False which is expected behavior.

File: rules/okta_rules/okta_new_behavior_accessing_admin_console.py [hunk 27]
```diff
@@ -0,0 +1,41 @@
+from panther_base_helpers import deep_get, get_val_from_list, okta_alert_context
+
+
+def rule(event):
+    if event.get("eventtype") != "policy.evaluate_sign_on":
```
[the remark this answers is not in the document]
Comment (arielkr256) [reply rules/okta_rules/okta_new_behavior_accessing_admin_console.py:5]: Same comment re: default values for human readable strings vs boolean logic

File: rules/okta_rules/okta_new_behavior_accessing_admin_console.py (hunk 20 above)
Comment (arielkr256) [reply rules/okta_rules/okta_new_behavior_accessing_admin_console.py:32]: default strings in human readable titles makes sense, ty

File: rules/okta_rules/okta_org2org_creation_modification.py (hunk 21 above)
Comment (arielkr256) [reply rules/okta_rules/okta_org2org_creation_modification.py:11]: Default value of None is ok in boolean logic imo

File: rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.py (hunk 25 above)
Comment (arielkr256) [reply rules/okta_rules/okta_phishing_attempt_blocked_by_fastpass.py:5]: added default strings for human readable title, not necessary for boolean logic imo

**Review by grantjoy — APPROVED** [review]

**Pull request merged**